blog/_posts/2020-03-31-extensions.md

```diff
@@ -0,0 +1,38 @@
+---
+slug: more-customizable-tiltfiles-with-extensions
+date: 2020-03-31
+author: victor
+layout: blog
+title: "More customizable Tiltfiles with Extensions"
+subtitle: "Share Tiltfile code snippets with the Tilt community in a streamlined platform. And get a free t-shirt!"
+image: "/assets/images/more-customizable-tiltfiles-with-extensions/michael-dziedzic-XTblNijO9IE-unsplash.jpg"
+image_caption: "Photo by <a href='https://unsplash.com/@lazycreekimages?utm_source=unsplash&utm_medium=referral&utm_content=creditCopyText'>Michael Dziedzic</a> on <a href='https://unsplash.com/?utm_source=unsplash&utm_medium=referral&utm_content=creditCopyText'>Unsplash</a>"
+tags:
+  - tilt  
+keywords:
+  - extensions
+  - tilt
+  - tiltfile
+---
+
+Since [releasing Tilt](/2018/08/28/how-tilt-updates-kubernetes-in-seconds-not-minutes.html), we’ve seen the Tilt community actively share code snippets of Tiltfile functionality in [Slack #tilt](https://kubernetes.slack.com/messages/CESBL84MV/), including ideas such as forcing resources into different namespaces, injecting sidecars, and even running tests. So we've created the open source Extensions platform to encourage this community collaboration in more structured fashion.
+
+Tilt Extensions is available starting in v0.12.11. [Upgrade](https://docs.tilt.dev/upgrade.html) to the latest version.
+
+## What's an extension, exactly? And how do you use it?
+An extension is simply a packaged function you call from within your Tiltfile. Simply import the extension (specifying the extension name and function name), and call it, all right inside your Tiltfile. 
```

Comment:
We use the word "simple" a lot in this blog post. Let's try to avoid it. For the audience we're writing for (infra engineers), they're going to be sensitive to overuse of words like "easy", "simple", "just", "straightforward", etc.

The old cliche of "show, don't tell" applies here.

One way to address this is to explain what we're trying to optimize for. e.g.,

```
You may have already defined a function in your Tiltfile!

def hi():
  print("hello world")

With Tilt Extensions, we wanted to create a clear path from:

- Take an existing a function in your Tiltfile
- Publish that function to a shared repository
- Replace the private function with the published version
````

Reply:
> We use the word "simple" a lot in this blog post. Let's try to avoid it. For the audience we're writing for (infra engineers), they're going to be sensitive to overuse of words like "easy", "simple", "just", "straightforward", etc.

I've removed all "simple"s in the blog post. 

> The old cliche of "show, don't tell" applies here.

I've simplified the explanation and relied on showing the "load" command.

> One way to address this is to explain what we're trying to optimize for. e.g.,

I don't think this directly applies in this section, since it is about using extensions, not taking an existing function and then publishing it. But I do think it's good content. I've moved a version of this content to the subsequent section about contributing an extension.